pkg/csi/service/vanilla/controller.go

```diff
@@ -487,9 +487,18 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 			}
 			if len(queryResult.Volumes) > 0 {
 				// Find datastore topology from the retrieved datastoreURL
+				if queryResult.Volumes[0].DatastoreUrl == "" {
```

Comment:
As this is CreateVolume call, shouldn't we try QueryVolume at least twice before we send out an error?

Reply:
We have cached the volume create tasks, so in the next call, volume will be retrieved from the cached VC task, and we will continue with queryvolume again.

Reply:
Okay makes sense then.